test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java

```diff
@@ -0,0 +1,785 @@
+		checkResults(src, ifVtArrayDst);
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			src[i] = new SomePrimitiveValueClass(i*25);
+		}
+		testOBJOBJ(src, ifPrimitiveVtArrayDst);
+		checkResults(src, ifPrimitiveVtArrayDst);
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy21() throws Throwable {
+
+		Object[] dst = new Object[ARRAY_SIZE];
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			dst[i] = new SomeIdentityClass(i*23);
+		}
+		testOBJOBJ(ifIdArraySrc, dst); // Fist invocation (Interpreter)
+
+		initArrays();
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			dst[i] = new SomeIdentityClass(i*24);
+		}
+		testOBJOBJ(ifIdArraySrc, dst);
+		checkResults(ifIdArraySrc, dst);
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			dst[i] = new SomeValueClass(i*25);
+		}
+		testOBJOBJ(ifVtArraySrc, dst);
+		checkResults(ifVtArraySrc, dst);
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			dst[i] = new SomePrimitiveValueClass(i*26);
+		}
+		testOBJOBJ(ifPrimitiveVtArraySrc, dst);
+		checkResults(ifPrimitiveVtArraySrc, dst);
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy22() throws Throwable {
+
+		try {
+			initArraysForNPETest(); // ifArray3[ARRAY_SIZE/2] is NULL
+			testIFPVT(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.NullPointerException npe1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForNPETest();
+				testIFPVT(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.NullPointerException npe2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
+			}
+		}
+
+		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy23() throws Throwable {
+
+		try {
+			initArraysForNPETest(); // ifArray3[ARRAY_SIZE/2] is NULL
+			testIFIF(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.NullPointerException npe1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForNPETest();
+				testIFIF(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.NullPointerException npe2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
+			}
+		}
+
+		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy24() throws Throwable {
+
+		try {
+			initArraysForArrayStoreChkExceptionTest();  // ifArray3[ARRAY_SIZE/2] is SomeIdentityClass
+			testIFPVT(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.ArrayStoreException ase1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForArrayStoreChkExceptionTest();
+				testIFPVT(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.ArrayStoreException ase2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
```

Comment:
This `return` statement needs one more tab to line up with the rest of the code in this block.